app/src/main/java/kth/jjve/xfran/CalendarViewActivity.java

```diff
@@ -111,22 +114,28 @@ private void setWeekView(LocalDate date) {
         // set the adapter using the days that are in the particular week
         calendarRecyclerView.setAdapter(new CalendarAdapter(daysInWeekArray, this));
 
-        // clear the list of planned workouts
-        eventListView.setAdapter(new EventAdapter(getApplicationContext(), new ArrayList<>()));
+        eventListView.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
+        eventListView.setAdapter(new EventAdapter(getApplicationContext(), new ArrayList<>(), this::onAddResult));
 
         // using the view model
         EventVM eventVM = ViewModelProviders.of(this).get(EventVM.class);
         eventVM.init(CalendarUtils.selectedDate);
         eventVM.getEvents().observe(this, this::setRecyclerView);
     }
 
+    public void onAddResult(int position) {
+        Intent intent = new Intent(this, WorkoutsListActivity.class);
```

Comment:
if you are making a new coomit anyways, you could consider writing this in one line. Otherwise, don't bother